deal_thief/conftest.py

```diff
@@ -0,0 +1,106 @@
+"""Conftest file the does the prep work prior tests run."""
+import pytest
+from passlib.apps import custom_app_context as pwd_context
+from pyramid import testing
+from .models import (
+        User,
+        Search,
+        get_engine,
+        get_session_factory,
+        get_tm_session
+)
+from .models.meta import Base
+import transaction
+
+
+class MockRequest(object):
+    """MockRequest intended to be a better version of testing.DummyRequest."""
+
+    def __init__(self, **kwargs):
+        """Init a MockRequest instance."""
+        self.__dict__.update(kwargs)
+
+    def route_url(self, string):
+        """Mimic the pyramid.request.Request route_url."""
+        return '/{}'.format(string)
```

Comment:
You can get the appropriate behavior of `request.route_url` by ensuring that your `.routes` module configuration is included properly by the config returned by a call to `testing.setUp`.  I'll point out specifically where in another comment below.  The point is that this work you've done here is good, but not needed.